pkg/oci/types.go

```diff
@@ -61,6 +61,7 @@ func (e *ArtifactType) Type() string {
 type RegistryResult struct {
 	Digest string
 	Config ArtifactConfig
+	Type   ArtifactType
```

Comment:
Should we add the platform, for the artifacts that support it, to the `RegistryResult`? If yes it should be easy to add such information when pulling artifacts.

Reply:
`os` and `arch` are params of the `Pull` function and also for the `Push` we have a way to specify them with options, so the caller already knows which platform is going to be used for artifacts that support it. If a wrong platform is passed for example in the pull operation, we are printing out this error: `no matching manifest was found in the manifest list` deriving from the `SelectManifest` function in oras